### PR TITLE
[14.0][ADD] endpoint_product_catalog

### PR DESCRIPTION
**Pull request by SilvioC2C**
Allows configuring specific endpoints to expose products catalogs.

### CODE REVIEW

File: endpoint_product_catalog/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2024 Camptocamp SA (<https://camptocamp.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Endpoint Product Catalog",
+    "version": "14.0.1.0.0",
+    "author": "Camptocamp, Odoo Community Association (OCA)",
+    "category": "Product",
+    "maintainers": ["SilvioC2C"],
```
Comment (simahawk) [inline endpoint_product_catalog/__manifest__.py]:
```suggestion
    "maintainers": ["SilvioC2C", "simahawk"],
```

File: oca_dependencies.txt [hunk 2]
```diff
@@ -1,3 +1,4 @@
 # See https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#oca_dependencies-txt
 reporting-engine
 server-ux
+web-api
```
Comment (simahawk) [inline oca_dependencies.txt:4]: no need to touch this file but... yeah... maybe somebody is still using it.
Although I'm not sure this is the right place for this module (the repo I mean). Maybe "PIM" is more appropriate even if not directly bound.

File: endpoint_product_catalog/models/ir_filters.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+            rec.endpoint_count = len(rec.endpoint_ids)
+
+    def button_display_endpoints(self):
+        self.ensure_one()
+        xmlid = "endpoint.endpoint_endpoint_act_window"
+        return dict(
+            self.env["ir.actions.act_window"]._for_xml_id(xmlid),
+            name=_("Endpoints"),
+            domain=[("product_assortment_id", "=", self.id)],
+            context=dict(self._context, default_product_assortment_id=self.id),
```
Comment (simahawk) [inline endpoint_product_catalog/models/ir_filters.py]:
```suggestion
            context=dict(self.env,context, default_product_assortment_id=self.id),
```

File: endpoint_product_catalog/views/ir_filters.xml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+            ref="product_assortment.product_assortment_view_form"
+        />
+        <field name="arch" type="xml">
+            <div name="buttons" position="inside">
+                <button
+                    class="oe_stat_button"
+                    name="button_display_endpoints"
+                    icon="fa-star"
+                    type="object"
+                    attrs="{'invisible': [('endpoint_count', '=', 0)]}"
```
Comment (simahawk) [inline endpoint_product_catalog/views/ir_filters.xml]: this should be invisible if this is not an assortment filter

File: endpoint_product_catalog/models/ir_filters.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+    endpoint_ids = fields.One2many("endpoint.endpoint", "product_assortment_id")
+    endpoint_count = fields.Integer(compute="_compute_endpoint_count")
+
+    @api.depends("endpoint_ids")
+    def _compute_endpoint_count(self):
+        for rec in self:
+            rec.endpoint_count = len(rec.endpoint_ids)
+
+    def button_display_endpoints(self):
```
Comment (simahawk) [inline endpoint_product_catalog/models/ir_filters.py]:
```suggestion
    def action_view_endpoints(self):
```

**Review by simahawk — CHANGES_REQUESTED** [review]

File: endpoint_product_catalog/views/ir_filters.xml (hunk 4 above)
Comment (SilvioC2C) [reply endpoint_product_catalog/views/ir_filters.xml]: Well, `endpoint_count` is always 0 for non-assortment filters, so it's implicit. But yeah, I'll make it explicit.

File: oca_dependencies.txt (hunk 2 above)
Comment (SilvioC2C) [reply oca_dependencies.txt:4]: Should I move it to the PIM repo?

Comment (simahawk) [reply oca_dependencies.txt:4]: I don't know to be honest. Maybe a new repo like `web-api-contrib` (similar to `mis-builder-contrib`) would be better... 

@sbidoul @lmignon @etobella any opinion?

Comment (lmignon) [reply oca_dependencies.txt:4]: I don't really have an opinion, as I don't see the immediate benefit of such a module. In my mind, exposing an API means exposing a contract and the system must at all times ensure that the data is validated against the contract while being able to provide documentation detailing the terms of the contract. 
Apart from this comment, the `web-api` repo would probably be more appropriate, or a new `web-api-contrib` repo.

Comment (simahawk) [reply oca_dependencies.txt:4]: Validation and documentation will come. ATM is not fundamental and we need some freedom here :wink:

Comment (simahawk) [reply oca_dependencies.txt:4]: I'm proposing the new repo here https://github.com/OCA/repo-maintainer-conf/pull/27

File: endpoint_product_catalog/readme/USAGE.rst [hunk 6]
```diff
@@ -0,0 +1,16 @@
+This module by itself contains no business logic, but its configuration can be used to
+create endpoints' code snippets.
+
+For example:
+
+.. code-block:: python
+
+    prod_data = []
+    prod_domain = endpoint.product_assorment_id._get_eval_domain()
```
Comment (bealdav) [inline endpoint_product_catalog/readme/USAGE.rst:9]:
```suggestion
    prod_domain = endpoint.product_assortment_id._get_eval_domain()
```

**Comment by simahawk** [comment]
moved to https://github.com/OCA/web-api-contrib/pull/1

File: endpoint_product_catalog/readme/USAGE.rst (hunk 6 above)
Comment (simahawk) [reply endpoint_product_catalog/readme/USAGE.rst:9]: fixed, tnx

**Pull request closed without merging**